src/WebJobs.Script.WebHost/Controllers/SwaggerController.cs

```diff
@@ -30,77 +30,46 @@ public SwaggerController(ISwaggerDocumentManager swaggerDocumentManager, WebScri
         [Route("admin/host/swagger/default")]
         public IHttpActionResult GetGeneratedSwaggerDocument()
         {
-            try
-            {
-                _traceWriter.Verbose(Resources.SwaggerGenerateDocument);
-                var swaggerDocument = _swaggerDocumentManager.GenerateSwaggerDocument(_scriptHostManager.HttpFunctions);
-                return Ok(swaggerDocument);
-            }
-            catch (Exception ex)
-            {
-                _traceWriter.Error(Resources.SwaggerGenerateError, ex);
-                return InternalServerError();
-            }
+            _traceWriter.Verbose(Resources.SwaggerGenerateDocument);
+            var swaggerDocument = _swaggerDocumentManager.GenerateSwaggerDocument(_scriptHostManager.HttpFunctions);
+            return Ok(swaggerDocument);
         }
 
         [HttpGet]
         [Route("admin/host/swagger")]
         public async Task<IHttpActionResult> GetSwaggerDocumentAsync()
         {
             IHttpActionResult result = NotFound();
-            try
+            if (_scriptHostManager.Instance.ScriptConfig.SwaggerEnabled)
```

Comment:
Right now the API's for managing / generating swagger document are only accessible until SwaggerEnabled is true, this may change though. It was mainly kept open so that we could allow the customer to make all the changes to the swagger until it is ready to be published.